src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -3267,6 +3290,199 @@ public void testOomScoreAdj() throws Exception {
+    assertThat(service.spec().mode().replicated().replicas(), is(4L));
+
+    // update service with same spec, but bump the number of replicas by 1
+    sut.updateService(response.id(), service.version().index(), ServiceSpec.builder()
+            .withName(service.spec().name())
+            .withTaskTemplate(service.spec().taskTemplate())
+            .withServiceMode(ServiceMode.withReplicas(5))
+            .withEndpointSpec(service.spec().endpointSpec())
+            .withUpdateConfig(service.spec().updateConfig())
+            .build());
+    service = sut.inspectService(response.id());
+    assertThat(service.spec().mode().replicated().replicas(), is(5L));
+  }
+
+
+  @Test
+  public void testListServices() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+    List<Service> services = sut.listServices();
+    assertThat(services, is(empty()));
+
+    final ServiceSpec spec = createServiceSpec(randomName());
+
+    sut.createService(spec, new ServiceCreateOptions());
+
+    services = sut.listServices();
+    assertThat(services.size(), is(1));
+  }
+
+  @Test
+  public void testListServicesFilterById() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+    final ServiceSpec spec = createServiceSpec(randomName());
+    final ServiceCreateResponse response = sut.createService(spec, new ServiceCreateOptions());
+
+    final List<Service> services = sut
+        .listServices(Service.find().withServiceId(response.id()).build());
+    assertThat(services.size(), is(1));
+    assertThat(services.get(0).id(), is(response.id()));
+  }
+
+  @Test
+  public void testListServicesFilterByName() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+    final String serviceName = randomName();
+    final ServiceSpec spec = createServiceSpec(serviceName);
+    sut.createService(spec, new ServiceCreateOptions());
+
+    final List<Service> services =
+            sut.listServices(Service.find().withServiceName(serviceName).build());
+    assertThat(services.size(), is(1));
+    assertThat(services.get(0).spec().name(), is(serviceName));
+  }
+
+  @Test
+  public void testRemoveService() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+
+    final ServiceSpec spec = createServiceSpec(randomName());
+    final ServiceCreateResponse response = sut.createService(spec, new ServiceCreateOptions());
+    assertThat(sut.listServices(), is(not(empty())));
+    sut.removeService(response.id());
+    assertThat(sut.listServices(), is(empty()));
+  }
+
+  @Test
+  public void testInspectTask() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+
+    final ServiceSpec spec = createServiceSpec(randomName());
+    assertThat(sut.listTasks().size(), is(0));
+    final ServiceCreateResponse response = sut.createService(spec, new ServiceCreateOptions());
+    Thread.sleep(2000); // to give it a while to spin containers
```

Comment:
I'm not a fan of these sleeps. We should use something like https://github.com/awaitility/awaitility, but since we have them littered throughout the tests already, we can do that in another PR.